scripts/commands/checkOrphanPages.ts

```diff
@@ -0,0 +1,170 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+
+import { globby } from "globby";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+
+interface Arguments {
+  [x: string]: unknown;
+  currentApis: boolean;
+  devApis: boolean;
+  historicalApis: boolean;
+}
+
+// The steps to this script are:
+//  ✅ 1. Enter each folder within docs
+//  ✅ 2. Create a list of files in the _toc
+//  ✅ 3. Collect the list of files in directory
+//  ✅ 4. Check if each file in directory is in _toc
+
+// Needed features:
+//  ✅ - Check api docs if desired
+//   - Include command in .github/workflows/main.yml
+const readArgs = (): Arguments => {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("current-apis", {
+      type: "boolean",
+      default: false,
+      description: "Check the links in the current API docs.",
+    })
+    .option("dev-apis", {
+      type: "boolean",
+      default: false,
+      description: "Check the links in the /dev API docs.",
+    })
+    .option("historical-apis", {
+      type: "boolean",
+      default: false,
+      description:
+        "Check the links in the historical API docs, e.g. `api/qiskit/0.44`. " +
+        "Warning: this is slow.",
+    })
+    .parseSync();
+};
+
+async function main() {
+  const args = readArgs();
+
+  // DETERMINE THE TOC FILES TO CHECK AGAINST
+  const tocFiles = await determineTocFiles(args);
+
+  // SET PASS/FAIL STATUS
+  let allGood = true;
+  const orphanPages = [];
+
+  for (tocFile of tocFiles) {
+    // LOAD JSON CONTENTS OF TOC FILE
+    console.log("Checking toc in:", tocFile);
+    const jsonFileContents = await fs.readFile(tocFile, "utf-8");
+    const children = JSON.parse(jsonFileContents).children;
+    const files: [string] = [];
+
+    // COLLECT ALL URLS REFERENCED IN TOC FILE
+    const fileContents = await collectTocFileContents(children);
+    const flatFileContents = flattenArray(fileContents);
+    flatFileContents[0] = flatFileContents[0] + "/index";
```

Comment:
This would be a good helper function: `getTocUrls(filePath: string): Set<string>`. It will make this `main` function easier to read.

The helper functions you already have are good. Keep those. This new helper function will still call those.